Web updated design and shortcut

shadyan opened the pull request:
**Describe the pull request**

Updated the Web Interface design (UI) for Abbott's commands and added some convenient functionality (UX)

**Why**

To make the web interface look more roomy (kind of objective) and overall look better (strictly ~~objective~~ subjective)

Also added some convenient keyboard shortcuts

**How**

Please read the commit messages for additional details:

- Update Web Interface UI to be more roomy and uniform
- Add special UI for the case where the results are 0
- Add some breathing room near the bottom of the page
- Add number of found commands to the results
- Add global slash (/) keyboard shortcut to focus on the input
- Add ability to tab through command names

**Screenshots**
Default view:  
![image](https://github.com/AdamLearns/Abbott/assets/11218993/87843093-4d95-4537-978f-b6bbb629e2c7)

Limited results:  
![image](https://github.com/AdamLearns/Abbott/assets/11218993/fad6f22a-9a22-4303-8524-571c28ac8901)

No results:  
![image](https://github.com/AdamLearns/Abbott/assets/11218993/3bf8bed4-7aa4-4591-9a1e-1bf8fea6d190)

HiDeoo reviewed (commented):
Thanks for the contribution :raised_hands: 
 
Just got a notification for it, I don't even remember subscribing to that repo but I was already on GitHub so I thought I'd take a quick look and leave a review.

Disclaimer: 

- It's up to Adam to decide if he wants to merge this PR or not.
- I would definitely not address any of my feedback before it's get reviewed by Adam.
- I only skimmed through the code very quickly and did not test anything, check responsiveness, etc.

Comments:

- I think it would make more sense to check for [`event.key`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) instead of [`event.code`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code) for the search keybinding in this specific case.
- I would be nice to support `Ctrl + K` as well as `Cmd + K` for the search keybinding.
- Saw some CSS transitions, could be neat to prefix them with [`motion-safe`](https://tailwindcss.com/docs/transition-property#prefers-reduced-motion) to respect user's preference for reduced motion.
- What is the reasoning behind making command names tabbable? As it relies on adding `tabindex` to headings, this is usually discouraged as tab key navigation should be reserved for interactive elements which is not the case here. More info [here](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-noninteractive-tabindex.md).

shadyan commented:
Thanks for the feedback @HiDeoo.

I completely realize that many of these changes are a matter of subjective preference, so I don't mind if Adam decides not to use any of what's here for that or any other reason.

Here are some notes on your comments:
- Updated. I often use event.code for modifiers and named keys (like Enter), but I agree that event.key would be much better here. For example "?" would also previously match, because Shift + Slash would still be Slash. Good catch.
- I actually had a similar thought at the beginning but decided against it for 2 reasons:
  1. That shortcut often clashes with the default browser shortcut
  2. That shortcut is more commonly used for a Command Palette element, which typically is a separate modal/window with a list of searchable actions/commands. Perhaps Adam wants to add other shortcuts/actions to the page, and they would be displayed in such a command palette.
  Still, I'm not opposed to the idea, as a command palette does make sense when searching for commands, which is the entire purpose of this page
  Using slash for focusing on a search element is pretty common and I use it all the time.
- Great suggestion. Updated.
- Reasoning is composed of 2 points:
   1. I wanted a way to navigate up/down one command at a time using the keyboard
   2. I was considering adding a "Copy Link" button next to each command and putting a tab index on those buttons instead. This would allow people, particularly on mobile devices, to copy the direct link for each command with a single click, so without having to copy from the URL bar. I ended up scrapping this idea, though.
   I think it's hopefully alright to keep this as is, since the tab index is added to each command only once (and to the command name, specifically). Having said that, I'm open to removing if it's a major accessibility impediment.
